File: lab07_stack/lab07_task01/scripts/lab07_task01_node.py
#!/usr/bin/env python

import rospy
import rospkg
import rosbag
from trajectory_msgs.msg import JointTrajectory, JointTrajectoryPoint
from lab06_task02.youbotKine import youbot_kinematic

def youbot_traj():

    rospy.init_node('lab7youbot_traj')

    rospack = rospkg.RosPack()

    path = rospack.get_path('lab07_example01')

    ##TODO: Initialise the bag and read the topic "my_transform" from the data
    ##TODO: Do the inverse kinematic to those transform to get the corresponding joints.
    ##TODO: Publish the JointTrajetory message to the corresponding topic to make the robot moves.


    rospy.sleep(5)


if __name__ == '__main__':
    try:
        youbot_traj()
    except rospy.ROSInterruptException:
        pass

File: lab07/lab07task01/src/lab07task01_node.cpp
#include "ros/ros.h"
#include "rosbag/bag.h"
#include "rosbag/view.h"
#include "trajectory_msgs/JointTrajectory.h"
#include "sensor_msgs/JointState.h"

#include "lab06task02/youbotKine.h"

#include "boost/foreach.hpp"


int main(int argc, char **argv)
{

    ros::init(argc, argv, "lab7youbot_traj");

    rosbag::Bag mybag;

    trajectory_msgs::JointTrajectory my_traj;
    trajectory_msgs::JointTrajectoryPoint my_pt;


    //TODO: Initialise the bags and read the topic "my_transform" from the data
    //TODO: Do the inverse kinematic to those transform to get the corresponding joint.
    //TODO: Publish the JointTrajectory message to the corresponding topic to make the robot moves.

    //This sleep varies in different machines. This is to prevent the publisher destroying itself before even publishes one trajectory.
    sleep(5);

    mybag.close();
    return 5;
}
